core/xml.js

```diff
@@ -365,7 +365,14 @@ Blockly.Xml.textToDom = function(text) {
   var doc = Blockly.utils.xml.textToDomDocument(text);
   if (!doc || !doc.documentElement ||
       doc.getElementsByTagName('parsererror').length) {
-    throw Error('textToDom was unable to parse: ' + text);
+    var oParser = new DOMParser();
+    // Attempt to parse as html to deserialize control chars.
+    doc = oParser.parseFromString(text, 'text/html');
+    if (!doc || !doc.body.firstChild ||
+        doc.body.firstChild.nodeName.toLowerCase() != 'xml') {
```

Comment:
Note that Blockly.Xml.textToDom has a different way of detecting the same thing.  Can you choose one method and make them consistent?  Maybe even factor it out as a util function if that makes sense.  Your call, just pointing it out.

Reply:
Follow up on the failure detection, does this even work?  When I execute the following valid XML:
(new DOMParser).parseFromString('<xml xmlns=\"https://developers.google.com/blockly/xml\">ab</xml>', 'text/xml').body
The result is always null.  There doesn't seem to be a .body property on an XML.

Furthermore, when I execute the following invalid XML ad skip the body:
(new DOMParser).parseFromString('<xml xmlns=\"https://developers.google.com/blockly/xml\">a\u0001b</xml>', 'text/xml').firstChild
I get an 'xml' node, that contains a 'parsererror' node.

So it looks like this method of detecting an error is broken in two different ways.  Assuming that's correct, the follow-up question would be why do the tests pass?

Also, I think that's why the equivalent test in Blockly.Xml.textToDom checks for the existence of a 'parsererror' node.

Reply:
It looks like github cut off the relevant top part of this function haha. The whole thing looks like this:
```javascript
Blockly.Xml.textToDom = function(text) {
  var doc = Blockly.utils.xml.textToDomDocument(text);
  if (!doc || !doc.documentElement ||
      doc.getElementsByTagName('parsererror').length) {
    var oParser = new DOMParser();
    // Attempt to parse as HTML to deserialize control chars.
    doc = oParser.parseFromString(text, 'text/html');
    if (!doc || !doc.body.firstChild ||
        doc.body.firstChild.nodeName.toLowerCase() != 'xml') {
      throw Error('DOMParser was unable to parse: ' + text);
    }
    return /** @type {!Element} */ (doc.body.firstChild);
  }
  return doc.documentElement;
};
```

So the first bit that uses `'text/xml'` parsing does check for a `parsererror`. But that doesn't work for HTML parsing because it doesn't (seem to?) give you parser errors. For example if I run `oParser.parseFromString('<xml</xml>', 'text/html');` It gives me:
```xml
<html>
  <head></head>
  <body>
    <xml< xml=""></xml<>
  </body>
</html>
```
So we have to run that other weird `nodeName` check instead.

Also note that I wanted to add this logic to `Blockly.utils.xml.textToDomDocument`, but the HTML parsing results in an `Element` and not a `Document`, which would be a breaking change on a public API.